app/models/miq_server/server_smart_proxy.rb

```diff
@@ -89,14 +89,13 @@ def queue_call(ost)
       end
       $log.debug "#{log_prefix}: queuing call to #{self.class.name}##{ost.method_name}"
       # Queue call to scan_metadata or sync_metadata.
-      MiqQueue.put(
+      MiqQueue.submit_job(
+        :service     => "smartproxy",
         :class_name  => self.class.name,
         :instance_id => id,
         :method_name => ost.method_name,
         :args        => ost,
-        :server_guid => guid,
-        :role        => "smartproxy",
-        :queue_name  => "smartproxy",
+        :server_guid => guid, # this should be derived in service. fix this
```

Comment:
This ensures the SmartProxy code is run on the same server the scan job was selected to run since the selection logic involves several checks, like proxy affinity.

It is called from 
https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/scanning_mixin.rb#L135
and
https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/scanning_mixin.rb#L158